src/pycram/orm/base.py

```diff
@@ -93,73 +109,77 @@ def reset(cls):
-        super().__init__()
-        self.x = x
-        self.y = y
-        self.z = z
-        self.metadata_id = metadata_id
+    # def __init__(self, x: int, y: int, z: int, metadata_id: Optional[int] = None):
+    #     super().__init__()
+    #     self.x = x
+    #     self.y = y
+    #     self.z = z
+    #     self.metadata_id = metadata_id
 
 
-class Quaternion(Base):
+class Quaternion(MappedAsDataclass, Base):
     """ORM Class for Quaternions."""
 
     __tablename__ = "Quaternion"
 
-    x = sqlalchemy.Column(sqlalchemy.types.Float)
-    y = sqlalchemy.Column(sqlalchemy.types.Float)
-    z = sqlalchemy.Column(sqlalchemy.types.Float)
-    w = sqlalchemy.Column(sqlalchemy.types.Float)
+    x: Mapped[float]
+    y: Mapped[float]
+    z: Mapped[float]
+    w: Mapped[float]
+    metadata_id: Mapped[Optional[int]] = mapped_column(ForeignKey("MetaData.id"), default=None)
 
-    def __init__(self, x: float, y: float, z: float, w: float,  metadata_id: Optional[int] = None):
-        super().__init__()
-        self.x = x
-        self.y = y
-        self.z = z
-        self.w = w
-        self.metadata_id = metadata_id
+    # def __init__(self, x: float, y: float, z: float, w: float, metadata_id: Optional[int] = None):
+    #     super().__init__()
+    #     self.x = x
+    #     self.y = y
+    #     self.z = z
+    #     self.w = w
+    #     self.metadata_id = metadata_id
 
 
-class Color(Base):
+class Color(MappedAsDataclass, Base):
     """ORM Class for Colors."""
 
     __tablename__ = "Color"
 
-    r = sqlalchemy.Column(sqlalchemy.types.Float)
-    g = sqlalchemy.Column(sqlalchemy.types.Float)
-    b = sqlalchemy.Column(sqlalchemy.types.Float)
-    alpha = sqlalchemy.Column(sqlalchemy.types.Float)
+    r: Mapped[float]
+    g: Mapped[float]
+    b: Mapped[float]
+    alpha: Mapped[float]
 
-    def __init__(self, r: float, g: float, b: float, alpha: float):
-        super().__init__()
-        self.r = r
-        self.g = g
-        self.b = b
-        self.alpha = alpha
+    # def __init__(self, r: float, g: float, b: float, alpha: float):
+    #     super().__init__()
+    #     self.r = r
+    #     self.g = g
+    #     self.b = b
+    #     self.alpha = alpha
 
 
-class RobotState(Base):
+class RobotState(MappedAsDataclass, Base):
     """ORM Representation of a robots state."""
 
     __tablename__ = "RobotState"
 
-    position = sqlalchemy.Column(sqlalchemy.types.Integer, sqlalchemy.ForeignKey("Position.id"))
+    position: Mapped[int] = mapped_column(ForeignKey("Position.id"), init=False)
+    position_table_entry: Mapped["Position"] = relationship(init=False)
     """The position of the robot."""
 
-    orientation = sqlalchemy.Column(sqlalchemy.types.Integer, sqlalchemy.ForeignKey("Quaternion.id"))
+    orientation: Mapped[int] = mapped_column(ForeignKey("Quaternion.id"), init=False)
+    orientation_table_entry: Mapped["Quaternion"] = relationship(init=False)
```

Comment:
The actual object quaternion should be in the quaternion field. The foreign key should have a name like quaternion_fk or something consistent.